pkg/operator2/operator.go

```diff
@@ -158,10 +158,16 @@ func (c *authOperator) Sync(obj metav1.Object) error {
 	}
 
 	if err := c.handleSync(operatorConfig); err != nil {
+		if statusErr := c.setFailingStatus(operatorConfig, "OperatorSyncLoopError", err.Error()); statusErr != nil {
+			glog.Errorf("error updating operator status: %s", statusErr)
+		}
+
 		return err
 	}
 
-	// TODO update states and handle ClusterOperator spec/status
+	if statusErr := c.setAvailableStatus(operatorConfig); statusErr != nil {
```

Comment:
This is wholly insufficient - ex: if the deployment fails to rollout we would be stating everything is fine.